### ISSUE TITLE
Full PHP8 typesafe injectable properties

### ISSUE THREAD

**Issue by runner78** [neos/flow-development-collection#2601]
### Description
At the moment this is not possible with Flow 7.2:

```php
class MonitoringCommandController extends CommandController
{
    #[Flow\Inject(lazy: true)]
    protected ?MyLazyService $myLazyService = null;

   #[Flow\Inject(lazy: false)]
    protected ?MyService $myService = null;

   /**
     * @var AnalyticsService|null
    */
   #[Flow\Inject(lazy: false)]
    protected ?MyAnotherService $myAnotherService = null;

  /*code */

}
```

- The first example does not work because Flow uses another type as a placeholder to load the type on demand
- The second (as well as the first) example does not work because of the missing type annotation.
- The third example does not work because of the "|null" in the type annotation

My suggestion would be make the property Injection more robust fit for PHP8+

**Comment by albe** [comment]
See also https://github.com/neos/flow-development-collection/issues/2114 - we need to change the way we build DependencyProxies

**Comment by runner78** [comment]
For classes with few dependencies, you can also work well with constructor injection, but this is not the case with larger classes.

**Comment by robertlemke** [comment]
I'll take a look when https://github.com/neos/flow-development-collection/pull/2701 is merged.

**Comment by albe** [comment]
https://github.com/neos/flow-development-collection/pull/2782 (included in 8.0) has made this work basically, with the limitation that typed properties are never lazy.
